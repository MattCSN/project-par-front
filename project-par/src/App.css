/* Global Styles */
body {
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Roboto', sans-serif;
    background-color: #f4f4f9;
    color: #333;
}

a {
    color: #007bff;
    text-decoration: none;
}

a:hover {
    text-decoration: underline;
}

/* Button Styles */
button, .p-button {
    background-color: #007bff;
    color: white;
    border: none;
    padding: 10px 20px;
    border-radius: 25px;
    cursor: pointer;
    font-size: 1rem;
    transition: background-color 0.3s ease;
}

button:hover, .p-button:hover {
    background-color: #0056b3;
}

button:disabled, .p-button:disabled {
    background-color: #d6d6d6;
    cursor: not-allowed;
}

/* Card Styles */
.p-card {
    background-color: white;
    border-radius: 12px;
    box-shadow: 0 2px 10px rgba(0, 0, 0, 0.1);
    overflow: hidden;
    transition: box-shadow 0.3s ease;
}

.p-card:hover {
    box-shadow: 0 4px 20px rgba(0, 0, 0, 0.2);
}

/* Accordion Styles */
.p-accordion .p-accordion-header {
    background-color: #f4f4f9;
    color: #333;
    border: none;
}

.p-accordion .p-accordion-content {
    background-color: #fff;
    color: #555;
    border: none;
}