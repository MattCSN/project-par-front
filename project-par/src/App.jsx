import React from 'react';
import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
import HomePage from './pages/HomePage';
import AboutUsPage from './pages/AboutUsPage';
import GolfDetailsPage from './pages/GolfDetailsPage';
import ContactUsPage from './pages/ContactUsPage';

import 'primereact/resources/themes/saga-blue/theme.css'; // Thème de PrimeReact
import 'primereact/resources/primereact.min.css'; // Styles de base de PrimeReact
import 'primeicons/primeicons.css';
import {PrimeReactProvider} from "primereact/api"; // Icônes de PrimeReact
import './styles/global.css';

const App = () => {
    return (
        <PrimeReactProvider>
            <Router>
                <Routes>
                    <Route path="/" element={<HomePage/>}/>
                    <Route path="/about-us" element={<AboutUsPage/>}/>
                    <Route path="/golfs/:id" element={<GolfDetailsPage/>}/>
                    <Route path="/contact-us" element={<ContactUsPage/>}/>
                </Routes>
            </Router>
        </PrimeReactProvider>
    );
};

export default App;