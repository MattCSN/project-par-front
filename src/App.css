@font-face {
  font-family: 'Inter';
  src: url('./assets/fonts/Inter/Inter-VariableFont_opsz,wght.ttf') format('truetype'),
  url('./assets/fonts/Inter/Inter-Italic-VariableFont_opsz,wght.ttf') format('truetype');
}

.global-container {
  width: 100vw;
  height: 100vh;
  display: flex;
  flex-direction: column;
  align-content: center;
}