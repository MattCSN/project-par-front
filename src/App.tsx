import './App.css';
import HomePage from './pages/HomePage/HomePage.tsx';
import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
import CourseDetails from "./pages/CourseDetails/CourseDetails.tsx";

function App() {
    return (
        <Router>
            <Routes>
                <Route path="/" element={<HomePage/>}/>
                <Route path="/course-details/:courseId" element={<CourseDetails/>}/>
            </Routes>
        </Router>
    );
}

export default App;