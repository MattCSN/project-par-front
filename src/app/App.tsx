import { useState } from "react";
import { useTranslation } from "react-i18next";

import reactLogo from "../assets/react.svg";
import viteLogo from "../assets/vite.svg";

import "./App.css";

function App() {
  const { t } = useTranslation();
  const [count, setCount] = useState(0);

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank" rel="noreferrer">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>
      <h1>PROJECT PAR</h1>
      <div className="card">
        <button
          onClick={() => {
            setCount(() => count + 1);
          }}
        >
          {`${t("button.counter")} ${count.toString()}`}
        </button>
        <p>
          Edit <code>src/App.tsx</code> and save to test HMR
        </p>
      </div>
      <p className="read-the-docs">
        Click on the Vite and React logos to learn more
      </p>
    </>
  );
}

export default App;
