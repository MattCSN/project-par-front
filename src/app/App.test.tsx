import { expect } from "vitest";

import { customRender } from "../tests/custom-render";

import App from "./App";

describe("App", (): void => {
  it("should render correctly and match the snapshot", async (): Promise<void> => {
    const view = await customRender(<App />);

    expect(view).toMatchSnapshot();
  });
});
