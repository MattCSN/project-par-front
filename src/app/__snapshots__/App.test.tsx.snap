// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`App > should render correctly and match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <a
          href="https://vitejs.dev"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="Vite logo"
            class="logo"
            src="/src/assets/vite.svg"
          />
        </a>
        <a
          href="https://react.dev"
          rel="noreferrer"
          target="_blank"
        >
          <img
            alt="React logo"
            class="logo react"
            src="/src/assets/react.svg"
          />
        </a>
      </div>
      <h1>
        Vite + React
      </h1>
      <h1>
        PROJECT PAR
      </h1>
      <div
        class="card"
      >
        <button>
          The count is 0
        </button>
        <p>
          Edit 
          <code>
            src/App.tsx
          </code>
           and save to test HMR
        </p>
      </div>
      <p
        class="read-the-docs"
      >
        Click on the Vite and React logos to learn more
      </p>
    </div>
  </body>,
  "container": <div>
    <div>
      <a
        href="https://vitejs.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="Vite logo"
          class="logo"
          src="/src/assets/vite.svg"
        />
      </a>
      <a
        href="https://react.dev"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="React logo"
          class="logo react"
          src="/src/assets/react.svg"
        />
      </a>
    </div>
    <h1>
      Vite + React
    </h1>
    <h1>
      PROJECT PAR
    </h1>
    <div
      class="card"
    >
      <button>
        The count is 0
      </button>
      <p>
        Edit 
        <code>
          src/App.tsx
        </code>
         and save to test HMR
      </p>
    </div>
    <p
      class="read-the-docs"
    >
      Click on the Vite and React logos to learn more
    </p>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
